refactor: move clippy lints to common file

stringhandler opened the pull request:
Description
---
* Move all clippy lints to a single file
* Update Contributing.md to inform new developers of process
* Replace `cargo clippy` with `cargo lints clippy` in CI workflow

Motivation and Context
---
To reduce duplicate code and the chance of missing lints on some projects. This (currently) requires developers to install and use `cargo-lints` and run `cargo lints clippy --all-targets` to get all errors.

I would have preferred a solution that did not require the installation of a different crate, but after much research and testing, there is no better (cleaner) way of doing this with only clippy and cargo.

How Has This Been Tested?
---
Ran `cargo lints clippy --all-targets`

Commit:
refactor: move clippy lints to common file

stringhandler commented:
I'm creating this PR now, which should fail clippy. Hopefully I can get some help fixing the individual clippy warnings

Cifko commented on lints.toml:
```diff
@@ -0,0 +1,70 @@
+    'clippy::range_plus_one',
+    'clippy::struct_excessive_bools',
+    # perhaps this is a bit harsh
+    'clippy::too_many_lines',
```
I think the default value of 100 is reasonable to have.

CjS77 commented:
I like this a lot. 
I personally hate seeing lint configs in source code files, so this really butters my scones.

sdbondi commented on Cargo.toml:
```diff
@@ -35,10 +35,18 @@ members = [
+#debug = true
+
+
+#![cfg_attr(not(debug_assertions), deny(unused_variables))]
```
Can remove this?

stringhandler replied:
yeah, added while I was trying stuff

Cifko committed:
chore: fix new clippy errors (#3952)

Description
---
Fixing all the new clippy errors.

How Has This Been Tested?
---
cargo lints clippy --all-targets

Cifko committed:
chore: make clippy pass (#3957)

Description
---
This is just to make clippy pass. I've marked some(most) of the clippy flags. Now I can go and enable them one by one. After this, I will merge `clippy-lockdown` to `development` and do the subsequent PRs on development branch.

stringhandler commented:
Continued by @Cifko in #3961
